lib/chef/provider/cron.rb

```diff
@@ -51,6 +67,9 @@ def load_current_resource
         @current_resource = Chef::Resource::Cron.new(@new_resource.name)
         @current_resource.user(@new_resource.user)
         @cron_exists = false
+        # Raise an execption if Chef can't read or update the crontab.
+        can_modify_cron?
```

Comment:
okay, sorry, it sounds we should walk that back then.

in that case i think it may be fine to go the other direction and not worry about decorating the exception.  if you do `shell_out!("crontab -u foobar")` and `foobar` doesn't exist you'll get a perfectly acceptable ShellOut error with output from crontab indicating that the user doesn't exist.  you're probably doing a lot of work worrying about validating with `getpwnam` that you don't need to bother with.